express-notes-api/index.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const jsonMethod = express.json();
+app.use(jsonMethod);
+
+const data = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const dataArray = [];
+  for (const key in data.notes) {
+    if (!data.notes[key]) {
+      res.status(200).json(dataArray);
+    } else {
+      dataArray.push(data.notes[key]);
+    }
+  }
+  res.status(200).json(dataArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const iD = req.params.id;
+  if ((Math.sign(iD) === -1) || (!Number.isInteger(parseInt(iD)))) {
+    res.status(400).send({ error: 'id must be a positive integer' });
```

Comment:
Requests like `GET /api/notes/45.123` (not an integer) and `GET /api/notes/0` (not positive) should also result in a `400 Bad Request` response. Test requests like these to verify they produce the desired output.